core/src/main/java/org/mskcc/cbio/portal/model/CanonicalGene.java

```diff
@@ -56,19 +57,39 @@
     private String type;
 
     public CanonicalGene(String hugoGeneSymbol) {
-        this(-1, hugoGeneSymbol);
```

Comment:
this seems convoluted. I suppose this is why there is the prior code for resolving by entrezgeneid *or* hugo using a single function call.